src/cocotb/triggers.py

```diff
@@ -939,28 +930,45 @@ async def _wait(self) -> "Combine":
         elif len(self._triggers) == 1:
             await self._triggers[0]
         else:
-            waiters: List[cocotb.task.Task[Any]] = []
-            e = _InternalEvent(self)
-            triggers = list(self._triggers)
+            waiters: List[cocotb.task.Task[object]] = []
+            done = _InternalEvent(self)
+            exception: Union[BaseException, None] = None
+
+            def on_done(
+                task: cocotb.task.Task[object],
+            ) -> None:
+                # have to check cancelled first otherwise exception() will throw
+                if task.cancelled():
+                    waiters.remove(task)
+                    if not waiters:
+                        done.set()
+                        return
```

Comment:
We won't be able to hit these lines until we get `CancelledError` working. I'll make an issue to come back and fix this up after `CancelledError` is working.

Reply:
I just added a comment to #3441.